src/provider/Provider.ts

```diff
@@ -447,6 +443,40 @@ export class Provider extends Instantiable {
     }
   }
 
+  public async computeResult(
+    jobId: string,
+    index: number,
+    destination: string,
+    account: Account
+  ): Promise<any> {
+    await this.getNonce(account.getId())
+    let signatureMessage = account.getId()
+    signatureMessage += jobId
+    signatureMessage += String(index)
+    signatureMessage += this.nonce
+    const signature = await this.createHashSignature(account, signatureMessage)
+    const path = this.getComputeResultEndpoint()
+      ? this.getComputeResultEndpoint().urlPath
+      : null
+    if (!path) return null
+    let consumeUrl = path
+    consumeUrl += `?jobId=${jobId}`
+    consumeUrl += `&index=${String(index)}`
+    consumeUrl += `&signature=${signature}`
+    consumeUrl += `&consumerAddress=${account.getId()}`
+
+    try {
+      !destination
+        ? await this.ocean.utils.fetch.downloadFileBrowser(consumeUrl)
+        : await this.ocean.utils.fetch.downloadFile(consumeUrl, destination, index)
+    } catch (e) {
+      this.logger.error('Error getting job result')
+      this.logger.error(e)
+      throw e
+    }
+    return destination
```

Comment:
what's the point of returning this? You have destination as a parameter and then it's never reassigned

Reply:
yeah, it's overkill.  any result !== null means that the download succeeded